umake/ui/cli/__init__.py

```diff
@@ -157,6 +157,60 @@ def mangle_args_for_default_framework(args):
     return result_args
 
 
+def list_frameworks(listing, detailled=True):
+    if listing == "all" or listing == "available":
+        # Sort the categories to prevent a random list at each new program execution
+        for category in sorted(BaseCategory.categories.values(), key=lambda cat: cat.prog_name):
+            # The category must have at least one framework
+            if len(category.frameworks) == 0:
+                continue
+
+            if detailled:
+                print("{}: {}".format(category.prog_name, category.description))
+            else:
+                print("{}:".format(category.prog_name))
+
+            # Sort the frameworks to prevent a random list at each new program execution
+            for framework in sorted(category.frameworks.values(), key=lambda fram: fram.prog_name):
+                # Framework doesn't be removal
+                if framework.only_for_removal:
+                    continue
+
+                if listing == "available":
+                    if not framework.is_installable:
+                        continue
+
+                if detailled:
+                    # Attribute 'description' of 'framework' contains if the category is default
+                    print("  {}: {} ".format(framework.prog_name, framework.description), end="")
```

Comment:
When you will format the output in the calling functions, some rules for i18n support to translate in multiple languages and clearer code:
* this implies to always preeced messages with _("foo bar baz") (note the _)
* you should never do:
```
a="foo"
b="bar"
print(a + b)
```
Note that it's similar to the form you are using:
```
print("foo", end="")
print("bar", end="")
```
* said differently, you prefer using string concatenation with {} form, which can be translated, like:
```
a="foo"
b="bar"
print(_("{} {}").format(a, b))
```
Of course, you can mix:
```
a="foo"
print(_("bar {}").format(a))
```
* Same, you should use "\n" for line breaks in the previous sentence you are using, you should tend to avoid print() alone (this may be needed in some occasion, like when you want to explicitely break a line between categories).